### PR TITLE
Update models to add subsidy cost columns

### PR DESCRIPTION
**Pull request by nmonga91**
**High level description:** 
Adding the necessary Original Loan Subsidy Cost columns to the various tables to facilitate surfacing these values for Loans.

**Technical details:** 
The `transaction_normalized` and `awards` Django models were updated to account for the subsidy cost columns. All matviews and associated Django models were updated also to incorporate the new fields. 

The following migrations were added (the ones that need to be faked have been marked as such):
- 0022_update_awards_add_total_subsidy_cost (FAKED, column already exists in the database)
- 0023_update_tx_norm_add_subsidy_cost

**Link to JIRA Ticket:** 
[DEV-485](https://federal-spending-transparency.atlassian.net/browse/DEV-485)
[DEV-488](https://federal-spending-transparency.atlassian.net/browse/DEV-488)
[DEV-506](https://federal-spending-transparency.atlassian.net/browse/DEV-506)

**The following are ALL required for the PR to be merged:**
- [ ] Backend review completed
- [x] Matview impact assessment completed
- [x] Frontend impact assessment completed (N/A)
- [x] Data validation completed (N/A)
- [x] API Performance evaluation completed and present:
    ```
    Timing changes:
        Before = N/A
        After = N/A
    
    Data volume changes:
        Before = 48764993 (universal_award_matview on staging)
        After = 48764993 (normal view with latest SQL on staging)
    ```

### CODE REVIEW

File: usaspending_api/database_scripts/matviews/functions_and_enums.sql [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-
+--
+--CREATE TYPE total_obligation_bins AS ENUM ('<1M', '1M..25M', '25M..100M', '100M..500M', '>500M');
+--CREATE TYPE award_type AS ENUM ('A', 'B', 'C', 'D', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11');
+--CREATE TYPE action_type AS ENUM ('b','H','S','C','L','J','W','V','N','Y','A','R',
+--                                 'a','X','D','B','M','G','F','c','K','P','T','E');
+--CREATE TYPE contract_pricing_type AS ENUM ('0','1','2','3','A','B','J','K','L','M',
+--                                           'NO','O','P','R','S','T','U','V','X','Y','Z');
+--
+--
```
Comment (alburde1) [inline usaspending_api/database_scripts/matviews/functions_and_enums.sql]: Is this no longer required?

Comment (nmonga91) [reply usaspending_api/database_scripts/matviews/functions_and_enums.sql]: It is required, good catch. Accidentally left commented. Will fix.

File: usaspending_api/database_scripts/matviews/universal_transaction_matview.sql [hunk 2]
```diff
@@ -25,11 +25,13 @@ SELECT
   "transaction_normalized"."award_id",
   "awards"."category" AS award_category,
   "awards"."total_obligation",
+  "awards"."total_subsidy_cost",
   obligation_to_enum("awards"."total_obligation") AS total_obl_bin,
   "awards"."fain",
   "awards"."uri",
   "awards"."piid",
-  "transaction_normalized"."federal_action_obligation",
+  COALESCE("transaction_normalized"."federal_action_obligation", 0) AS "federal_action_obligation",
```
Comment (tony-sappe) [inline usaspending_api/database_scripts/matviews/universal_transaction_matview.sql]: Will postgres maintain the correct type using coalesce?

Comment (nmonga91) [reply usaspending_api/database_scripts/matviews/universal_transaction_matview.sql]: `0` should be able to be interpreted as any valid numeric type, but we could explicitly cast to `NUMERIC(20, 2)` if you believe that would be more comprehensive.

Comment (tony-sappe) [reply usaspending_api/database_scripts/matviews/universal_transaction_matview.sql]: Might as well be safe. Do you want to modify all of the existing numeric types to (20,2)?

Comment (nmonga91) [reply usaspending_api/database_scripts/matviews/universal_transaction_matview.sql]: I can do that, will make that update now.

File: usaspending_api/database_scripts/matview_generator/universal_transaction_matview.json [hunk 3]
```diff
@@ -144,6 +145,13 @@
     }, {
       "name": "total_obl_bin",
       "columns": [{"name": "\"total_obl_bin\""}]
+    }, {
+      "name": "total_subsidy_cost",
+      "where": "\"total_subsidy_cost\" IS NOT NULL",
```
Comment (tony-sappe) [inline usaspending_api/database_scripts/matview_generator/universal_transaction_matview.json]: Will this ever be null?

Comment (nmonga91) [reply usaspending_api/database_scripts/matview_generator/universal_transaction_matview.json]: Yes it can be when dealing with non-loan type transactions.

**Pull request merged**